stdlib/universe/chande_momentum_oscillator.go

```diff
@@ -0,0 +1,348 @@
+	cache execute.TableBuilderCache
+
+	n       int64
+	columns []string
+}
+
+func NewChandeMomentumOscillatorTransformation(d execute.Dataset, cache execute.TableBuilderCache, spec *ChandeMomentumOscillatorProcedureSpec) *chandeMomentumOscillatorTransformation {
+	return &chandeMomentumOscillatorTransformation{
+		d:       d,
+		cache:   cache,
+		n:       spec.N,
+		columns: spec.Columns,
+	}
+}
+
+func (t *chandeMomentumOscillatorTransformation) RetractTable(id execute.DatasetID, key flux.GroupKey) error {
+	return t.d.RetractTable(key)
+}
+
+func (t *chandeMomentumOscillatorTransformation) Process(id execute.DatasetID, tbl flux.Table) error {
+	builder, created := t.cache.TableBuilder(tbl.Key())
+	if !created {
+		return fmt.Errorf("chande momentum oscillator found duplicate table with key: %v", tbl.Key())
+	}
+	cols := tbl.Cols()
+	doChandeMomentumOscillator := make([]bool, len(cols))
+	for j, c := range cols {
+		found := false
+		for _, label := range t.columns {
+			if c.Label == label {
+				if c.Type != flux.TInt && c.Type != flux.TUInt && c.Type != flux.TFloat {
+					return fmt.Errorf("cannot take chande momentum oscillator of column %s (type %s)", c.Label, c.Type.String())
+				}
+				found = true
+				break
+			}
+		}
+
+		if found {
+			mac := c
+			mac.Type = flux.TFloat
+			_, err := builder.AddCol(mac)
+			if err != nil {
+				return err
+			}
+			doChandeMomentumOscillator[j] = true
+		} else {
+			_, err := builder.AddCol(c)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return tbl.Do(func(cr flux.ColReader) error {
+		if cr.Len() == 0 {
+			return nil
+		}
+
+		for j, c := range cr.Cols() {
+			if !doChandeMomentumOscillator[j] {
+				switch c.Type {
+				case flux.TTime:
+					arr := cr.Times(j)
+					for i := int(t.n); i < arr.Len(); i++ {
+						if err := builder.AppendTime(j, execute.Time(arr.Value(i))); err != nil {
+							return err
+						}
+					}
+				case flux.TBool:
+					arr := cr.Bools(j)
+					for i := int(t.n); i < arr.Len(); i++ {
+						if err := builder.AppendBool(j, arr.Value(i)); err != nil {
+							return err
+						}
+					}
+				case flux.TString:
+					arr := cr.Strings(j)
+					for i := int(t.n); i < arr.Len(); i++ {
+						if err := builder.AppendString(j, arr.ValueString(i)); err != nil {
+							return err
+						}
+					}
+				case flux.TInt:
+					arr := cr.Ints(j)
+					for i := int(t.n); i < arr.Len(); i++ {
+						if err := builder.AppendInt(j, arr.Value(i)); err != nil {
+							return err
+						}
+					}
+				case flux.TUInt:
+					arr := cr.UInts(j)
+					for i := int(t.n); i < arr.Len(); i++ {
+						if err := builder.AppendUInt(j, arr.Value(i)); err != nil {
+							return err
+						}
+					}
+				case flux.TFloat:
+					arr := cr.Floats(j)
+					for i := int(t.n); i < arr.Len(); i++ {
+						if err := builder.AppendFloat(j, arr.Value(i)); err != nil {
+							return err
+						}
+					}
+				}
+			} else {
+				if err := t.do(int(t.n), cr, c, builder, j); err != nil {
```

Comment:
Instead of `else`, you should put a `continue` at the end of the above if block and then have this second part just stay here unindented. The `else` is unnecessary.

Reply:
Is there a reason why this is better?

Reply:
It reduces indentation which makes it easier to read. But the main reason is just because it's part of the Go style guide.

https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow